tests/lookup/tests.py

```diff
@@ -753,6 +754,49 @@ def test_chain_date_time_lookups(self):
         )
 
 
+class ObjectLookupTests(LookupTests):
+    def test_direct_lookup_init(self):
+        lookup = lookups.GreaterThan(F('id'), Value(5))
+        queryset = Article.objects.filter(lookup)
+        self.assertQuerysetEqual(queryset, ['<Article: Article 6>',  '<Article: Article 7>'])
+
+    def test_lookup_with_no_name(self):
+        class CustomGreaterThanLookup(lookups.GreaterThan):
+            lookup_name = None
+
+            def get_rhs_op(self, connect, rhs):
+                return '> {}'.format(rhs)
+
+        lookup = CustomGreaterThanLookup(F('id'), Value(5))
+        queryset = Article.objects.filter(lookup)
+        self.assertQuerysetEqual(queryset, ['<Article: Article 6>',  '<Article: Article 7>'])
+
+    def test_direct_lookup_with_transform(self):
+        lookup = lookups.Contains(Upper(Lower(Upper(F('headline')))), 'ARTICLE 5')
+        queryset = Article.objects.filter(lookup)
+        self.assertQuerysetEqual(queryset, ['<Article: Article 5>'])
+
+    def test_lookup_exclude(self):
+        lookup = lookups.LessThanOrEqual(F('id'), Value(5))
+        queryset = Article.objects.exclude(lookup)
+        self.assertQuerysetEqual(queryset, ['<Article: Article 6>',  '<Article: Article 7>'])
+
+    def test_lookup_exclude2(self):
+        season_2009 = Season.objects.create(year=2009, gt=111)
+        season_2009.games.create(home="Houston Astros", away="St. Louis Cardinals")
+
+        season_2010 = Season.objects.create(year=2010, gt=222)
+        season_2010.games.create(home="Houston Astros", away="Chicago Cubs")
+
+        # keyword-based lookup
+        games = Game.objects.exclude(season__year__gte=2010)
+        self.assertQuerysetEqual(games, ['<Game: St. Louis Cardinals at Houston Astros>'])
+
+        # object-based lookup
+        games = Game.objects.exclude(lookups.GreaterThanOrEqual(F('season__year'), Value(2010)))
+        self.assertQuerysetEqual(games, ['<Game: St. Louis Cardinals at Houston Astros>'])
```

Comment:
The interesting test case is this:
```
        season_2009 = Season.objects.create(year=2009, gt=111)
        season_2009.games.create(home="Houston Astros", away="St. Louis Cardinals")
        season_2009.games.create(home="Houston Astros", away="Chicago Cubs")
        season_2009.games.create(home="St. Louis Cardinals", away="Houston Atros")
        season_2010 = Season.objects.create(year=2010, gt=222)
        season_2010.games.create(home="Houston Astros", away="Chicago Cubs")
        qs1 = Season.objects.exclude(games__home__contains='Houston')
        qs2 = Season.objects.exclude(lookups.Contains(F('games__home'), 'Houston'))
```
where the qs1 and qs2 objects should match.

The encouraging thing is that the code is trying to call split_exclude().

It might be hard to make this test actually work, the split_exclude() code is pretty big hack, and especially if the expression creates joins to multiple different relations it will be hard to make the current code work properly. But, if I recall correctly, we don't support .filter(games__away__contains=F('games__home')) either.